src/bootstrap/builder.rs

```diff
@@ -919,7 +919,7 @@ impl<'a> Builder<'a> {
         host: TargetSelection,
         target: TargetSelection,
     ) -> Compiler {
-        if self.build.force_use_stage2() {
+        if self.build.force_use_stage2(Compiler { stage, host }) {
```

Comment:
I would prefer that we avoid creating Compiler structs if they don't correspond to an already-built compiler (i.e., the only constructor should be `self.compiler`).

Can we refactor both force_use functions to take stage and host as separate values?